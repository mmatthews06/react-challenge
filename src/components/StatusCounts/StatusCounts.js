import React from 'react';
import PropTypes from 'prop-types';
import { Grid } from '@material-ui/core';

import StatusCard from '../StatusCard/StatusCard';

export default function StatusCounts({ active }) {
  return (
    <Grid container>
      <Grid item>
        <StatusCard label="Active" value={active} />
      </Grid>
    </Grid>
  );
}

StatusCounts.propTypes = {
  active: PropTypes.number.isRequired,
};
