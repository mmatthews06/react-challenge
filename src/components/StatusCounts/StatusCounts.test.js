/* global describe, expect, it */
import React from 'react';
import { shallow } from 'enzyme';

import StatusCounts from './StatusCounts';

describe('StatusCard', () => {
  it('should match a snapshot', () => {
    expect(shallow(<StatusCounts active={20} queued={5} onHold={3} />)).toMatchSnapshot();
  });
});
