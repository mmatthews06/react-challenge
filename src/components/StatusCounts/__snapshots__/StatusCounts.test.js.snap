// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusCard should match a snapshot 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  spacing={10}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Status
      label="Active"
      value={20}
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Status
      label="Queued"
      value={5}
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Status
      label="On Hold"
      value={3}
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Status
      label="Total"
      value={28}
    />
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
